redirect/main.tf

```diff
@@ -12,15 +12,33 @@ locals {
 resource "aws_s3_bucket" "redirect" {
   count = var.create ? 1 : 0
 
-  bucket = var.bucket != null ? var.bucket : "${local.name_prefix}-redirect"
+  bucket = local.bucket
   acl    = "public-read"
   tags   = local.tags
 
   website {
-    redirect_all_requests_to = var.target
+    index_document = "index.html"
+    routing_rules  = <<EOF
+[{
+    "Redirect": {
+        "Protocol": "${var.protocol}",
+        "HostName": "${var.host}",
+        "HttpRedirectCode": "${var.redirect_status_code}"
+    }
+}]
+EOF
```

Comment:
you can use `<<-EOF` to keep the source code aligned
```tf
    routing_rules  = <<-EOF
      [{
        "Redirect": {
          "Protocol": "${var.protocol}",
          "HostName": "${var.host}",
          "HttpRedirectCode": "${var.redirect_status_code}"
        }
      }]
    EOF
```

https://www.terraform.io/docs/configuration/expressions.html#string-literals